### PR TITLE
Forms

### PR DESCRIPTION
**Pull request by slkendell-mt**

### CODE REVIEW

File: backoffice/src/Models/Domain/Droit.cs [hunk 1]
```diff
@@ -1,7 +1,59 @@
...
+    public int? WreckVesselYearSunk { get; set; }
+
+
+    // Location
+    public string? Latitude { get; set; }
+    public string? Longitude { get; set; }
+    public bool InUkWaters { get; set; } = false;
+    public int? LocationRadius { get; set; } //Units? Unknown.
+    public int? Depth { get; set; } //Units? Unknown. //metres from frontend
+    public string? LocationDescription { get; set; } = string.Empty;
```
Comment (Evie-Skinner18) [inline backoffice/src/Models/Domain/Droit.cs:38]: Is there are reason why the domain models look more like data structures than objects? Are we going to flesh them out with more behaviour as we learn more about the domain so they don't stay so anaemic?

File: backoffice/src/Models/Domain/Droit.cs [hunk 2]
```diff
@@ -1,7 +1,59 @@
...
+
+
+    // Salvage
+
+    public bool SalvageAwardClaimed { get; set; } = false;
+    public string? ServicesDescription { get; set; }
+    public int? ServicesDuration { get; set; } //Units? Unknown.
```
Comment (Evie-Skinner18) [inline backoffice/src/Models/Domain/Droit.cs:45]: Are we going to have domain models separate from data structures that we use e.g as API responses/entity models?

File: backoffice/src/Models/Domain/DroitItemOwner.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    public Guid DroitItemId { get; set; }
+
+    public DateTime? OwnerContactedDate { get; set; }
+    public DateTime? OwnerRespondedDate { get; set; }
+
+    public DateTime? OwnerReceiptsReceivedDate { get; set; }
+
+    public bool OwnerHasResponded => OwnerRespondedDate.HasValue;
+
+    public bool OwnerResponseWindowClosed => OwnerRespondedDate != null && !OwnerHasResponded &&
```
Comment (Evie-Skinner18) [inline backoffice/src/Models/Domain/DroitItemOwner.cs:17]: <3 this but I can see we're not unit testing it yet

File: backoffice/src/Services/Enums.cs [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+public enum EmailTemplateType
+{
+    ReportAcknowledged,
+    ReportCompleteInformation,
+    ReportConfirmed,
+    LloydsOfLondonNotification,
+    ClosureOwnerNotFound,
+    ClosureOwnerWaivesRights,
+    ClosureCustom,
+    CustomEmail
```
Comment (Evie-Skinner18) [inline backoffice/src/Services/Enums.cs:12]: There should be another template type here which we added in commit f6da5d2745958fd48253d25a702774bc96475e2d

File: backoffice/src/Views/Droit/Add.cshtml [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+<form class="form-horizontal mt-4"asp-action="Create" method="POST">
+    <div class="row row-cols-1 row-cols-md-2">
+    <div class="col mt-4">
+
+        <div class="card">
+            <h5 class="card-header">Droit Report</h5>
+            <div class="card-body">
+
+                @* <div class="form-group">
+                    @Html.LabelFor(m => m.Id, new {@class="col-form-label"})
```
Comment (Evie-Skinner18) [inline backoffice/src/Views/Droit/Add.cshtml:15]: Consider removing commented out code

File: backoffice/src/Views/Droit/Add.cshtml [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+                    @Html.LabelFor(m => m.ReportedDate, new { @class = "col-form-label" })
+                    @Html.EditorFor(m => m.ReportedDate, new { htmlAttributes = new { @class = "form-control" } })
+                </div>
+
+
+                @* <div class="form-group">
+                    @Html.LabelFor(m => m.Created, new {@class="col-form-label"})
+                    @Html.EditorFor(m => m.Created,  new { htmlAttributes = new {@class="form-control"
+                    , @Value = Model.Created != null ? Model.Created.ToString("yyyy-MM-dd") : DateTime.Now.ToString("yyyy-MM-dd") }})
+                </div> *@
```
Comment (Evie-Skinner18) [inline backoffice/src/Views/Droit/Add.cshtml:45]: Consider removing

File: backoffice/src/Views/Droit/Add.cshtml [hunk 7]
```diff
@@ -0,0 +1,223 @@
...
+                <div class="form-group">
+                    @Html.LabelFor(m => m.Reference, new { @class = "col-form-label" })
+                    @Html.EditorFor(m => m.Reference, new { htmlAttributes = new { @class = "form-control" } })
+                </div>
+
+                <div class="form-group">
+                    @Html.LabelFor(m => m.Status, new { @class = "col-form-label" })
+                    @Html.DropDownListFor(m => m.Status, new List<SelectListItem>
+                    {
+                        new() { Text = "Received", Value = "Received" },
```
Comment (Evie-Skinner18) [inline backoffice/src/Views/Droit/Add.cshtml:28]: <3 this use of the most up to date .Net syntax!

File: backoffice/src/Views/Droit/Index.cshtml [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+        <th scope="col">Created</th>
+    </tr>
+    </thead>
+    <tbody>
+    @foreach (var droit in Model)
+    {
+        <tr>
+            <th scope="row">
+                <a asp-action="View" asp-route-id="@droit.Id">@droit.Reference</a>
+            </th>
```
Comment (Evie-Skinner18) [inline backoffice/src/Views/Droit/Index.cshtml:21]: ❤️

File: backoffice/src/Models/Domain/Droit.cs (hunk 1 above)
Comment (slkendell-mt) [reply backoffice/src/Models/Domain/Droit.cs:38]: Yeah as they're unknowns ATM, makes sense to keep them as basic as possible. In theory these will probably stay quite lightweight as the entity framework models, then we'd expand them with functional methods, or have extension methods to keep them off the core model

File: backoffice/src/Models/Domain/Droit.cs (hunk 2 above)
Comment (slkendell-mt) [reply backoffice/src/Models/Domain/Droit.cs:45]: Not sure the best approach at the moment, but it's worth thinking about now

**Pull request merged**